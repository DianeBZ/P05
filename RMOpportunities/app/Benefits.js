import React, { Component } from 'react';
import { Text, Image, View,ScrollView, Dimensions, TouchableHighlight, BackAndroid} from 'react-native';

import Translation from './Translation';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

var valorize = 'false';
var price = 'false';
var time = 'false';
var visibility = 'false';
var environment = 'false';


class BlockFunction extends Component{

 render(){
     return(
       <View>
       {(() => {
           if (valorize==='true'){
             return(
               <View style={styles.containerV}>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressValoriserStocks}>
                     <View style = {styles.cliqueDecoupage}>
                       <View style = {styles.cliqueTitre}>
                         <Image source={require('../img/valoriser.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].val_stock}</Text>
                       </View>
                       <Text style = {styles.cliqueTexte}>{Translation[lang].text_val_stock}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressPrix}>
                     <View style = {styles.cliqueDecoupageStatique}>
                       <Image source={require('../img/prix.png')} style = {styles.logoBloque} />
                       <Text style = {styles.texteCarreau}>{Translation[lang].meill_prix}</Text>
                     </View>
                   </TouchableHighlight>
                 </View>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressTemps}>
                     <View style = {styles.decoupage}>
                       <Image source={require('../img/temps.png')} style = {styles.logoBloque} />
                       <Text style = {styles.texteCarreau}>{Translation[lang].gagner_tps}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressVisibilite}>
                   <View style = {styles.decoupage}>
                     <Image source={require('../img/visibilite.png')} style = {styles.logoBloque} />
                     <Text style = {styles.texteCarreau}>{Translation[lang].gde_visib}</Text>
                   </View>
                   </TouchableHighlight>
                 </View>
                 <TouchableHighlight onPress={this._onPressEnvironnement}>
                 <View style = {styles.decoupage}>
                   <Image source={require('../img/environnement.png')} style = {styles.logoBloque} />
                   <Text style = {styles.texteCarreau}>{Translation[lang].prot_env}</Text>
                 </View>
                 </TouchableHighlight>
               </View>
             );
           }else if(price === 'true'){
             return(
               <View style={styles.containerV}>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressValoriserStocks}>
                     <View style = {styles.cliqueDecoupageStatique}>
                         <Image source={require('../img/valoriser.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].val_stock}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressPrix}>
                     <View style = {styles.cliqueDecoupage}>
                       <View style = {styles.cliqueTitre}>
                         <Image source={require('../img/prix.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].meill_prix}</Text>
                       </View>
                       <Text style = {styles.cliqueTexte}>{Translation[lang].text_meill_prix}</Text>
                     </View>
                   </TouchableHighlight>
                 </View>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressTemps}>
                     <View style = {styles.decoupage}>
                       <Image source={require('../img/temps.png')} style = {styles.logoBloque} />
                       <Text style = {styles.texteCarreau}>{Translation[lang].gagner_tps}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressVisibilite}>
                   <View style = {styles.decoupage}>
                     <Image source={require('../img/visibilite.png')} style = {styles.logoBloque} />
                     <Text style = {styles.texteCarreau}>{Translation[lang].gde_visib}</Text>
                   </View>
                   </TouchableHighlight>
                 </View>
                 <TouchableHighlight onPress={this._onPressEnvironnement}>
                 <View style = {styles.decoupage}>
                   <Image source={require('../img/environnement.png')} style = {styles.logoBloque} />
                   <Text style = {styles.texteCarreau}>{Translation[lang].prot_env}</Text>
                 </View>
                 </TouchableHighlight>
               </View>
             );
           }else if(time === 'true'){
             return(
               <View style={styles.containerV}>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressValoriserStocks}>
                     <View style = {styles.decoupage}>
                         <Image source={require('../img/valoriser.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].val_stock}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressPrix}>
                     <View style = {styles.decoupage}>
                         <Image source={require('../img/prix.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].meill_prix}</Text>
                     </View>
                   </TouchableHighlight>
                 </View>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressTemps}>
                     <View style = {styles.cliqueDecoupage}>
                       <View style = {styles.cliqueTitre}>
                         <Image source={require('../img/temps.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].gagner_tps}</Text>
                       </View>
                       <Text style = {styles.cliqueTexte}>{Translation[lang].text_gagner_tps}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressVisibilite}>
                   <View style = {styles.cliqueDecoupageStatique}>
                     <Image source={require('../img/visibilite.png')} style = {styles.logoBloque} />
                     <Text style = {styles.texteCarreau}>{Translation[lang].gde_visib}</Text>
                   </View>
                   </TouchableHighlight>
                 </View>
                 <TouchableHighlight onPress={this._onPressEnvironnement}>
                 <View style = {styles.decoupage}>
                   <Image source={require('../img/environnement.png')} style = {styles.logoBloque} />
                   <Text style = {styles.texteCarreau}>{Translation[lang].prot_env}</Text>
                 </View>
                 </TouchableHighlight>
               </View>
             );
           }else if(visibilite === 'true'){
             return(
               <View style={styles.containerV}>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressValoriserStocks}>
                     <View style = {styles.decoupage}>
                         <Image source={require('../img/valoriser.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].val_stock}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressPrix}>
                     <View style = {styles.decoupage}>
                         <Image source={require('../img/prix.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].meill_prix}</Text>
                     </View>
                   </TouchableHighlight>
                 </View>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressTemps}>
                     <View style = {styles.cliqueDecoupageStatique}>
                       <Image source={require('../img/temps.png')} style = {styles.logoBloque} />
                       <Text style = {styles.texteCarreau}>{Translation[lang].gagner_tps}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressVisibilite}>
                   <View style = {styles.cliqueDecoupage}>
                     <View style = {styles.cliqueTitre}>
                       <Image source={require('../img/visibilite.png')} style = {styles.logoBloque} />
                       <Text style = {styles.texteCarreau}>{Translation[lang].gde_visib}</Text>
                     </View>
                     <Text style = {styles.cliqueTexte}>{Translation[lang].text_gde_visib}</Text>
                   </View>
                   </TouchableHighlight>
                 </View>
                 <TouchableHighlight onPress={this._onPressEnvironnement}>
                 <View style = {styles.decoupage}>
                   <Image source={require('../img/environnement.png')} style = {styles.logoBloque} />
                   <Text style = {styles.texteCarreau}>{Translation[lang].prot_env}</Text>
                 </View>
                 </TouchableHighlight>
               </View>
             );
           }else if(environment === 'true'){
             return(
               <View style={styles.containerV}>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressValoriserStocks}>
                     <View style = {styles.decoupage}>
                         <Image source={require('../img/valoriser.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].val_stock}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressPrix}>
                     <View style = {styles.decoupage}>
                         <Image source={require('../img/prix.png')} style = {styles.logoBloque} />
                         <Text style = {styles.texteCarreau}>{Translation[lang].meill_prix}</Text>
                     </View>
                   </TouchableHighlight>
                 </View>
                 <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressTemps}>
                     <View style = {styles.decoupage}>
                       <Image source={require('../img/temps.png')} style = {styles.logoBloque} />
                       <Text style = {styles.texteCarreau}>{Translation[lang].gagner_tps}</Text>
                     </View>
                   </TouchableHighlight>
                   <TouchableHighlight onPress={this._onPressVisibilite}>
                   <View style = {styles.decoupage}>
                     <Image source={require('../img/visibilite.png')} style = {styles.logoBloque} />
                     <Text style = {styles.texteCarreau}>{Translation[lang].gde_visib}</Text>
                   </View>
                   </TouchableHighlight>
                 </View>
                 <TouchableHighlight onPress={this._onPressEnvironnement}>
                 <View style = {[styles.cliqueDecoupage, {marginBottom: width *0.1}]}>
                   <View style = {styles.cliqueTitre}>
                     <Image source={require('../img/environnement.png')} style = {styles.logoBloque} />
                     <Text style = {styles.texteCarreau}>{Translation[lang].prot_env}</Text>
                   </View>
                   <Text style = {styles.cliqueTexte}>{Translation[lang].text_prot_env}</Text>
                 </View>
                 </TouchableHighlight>
               </View>
             );
           }else{
               return(
                 <View style={styles.containerV}>
                   <View style={styles.containerH}>
                     <TouchableHighlight onPress={this._onPressValoriserStocks}>
                       <View style = {styles.decoupage}>
                           <Image source={require('../img/valoriser.png')} style = {styles.logoBloque} />
                           <Text style = {styles.texteCarreau}>{Translation[lang].val_stock}</Text>
                       </View>
                     </TouchableHighlight>
                     <TouchableHighlight onPress={this._onPressPrix}>
                     <View style = {styles.decoupage}>
                       <Image source={require('../img/prix.png')} style = {styles.logoBloque} />
                       <Text style = {styles.texteCarreau}>{Translation[lang].meill_prix}</Text>
                     </View>
                     </TouchableHighlight>
                   </View>
                   <View style={styles.containerH}>
                   <TouchableHighlight onPress={this._onPressTemps}>
                     <View style = {styles.decoupage}>
                       <Image source={require('../img/temps.png')} style = {styles.logoBloque} />
                       <Text style = {styles.texteCarreau}>{Translation[lang].gagner_tps}</Text>
                     </View>
                     </TouchableHighlight>
                     <TouchableHighlight onPress={this._onPressVisibilite}>
                     <View style = {styles.decoupage}>
                       <Image source={require('../img/visibilite.png')} style = {styles.logoBloque} />
                       <Text style = {styles.texteCarreau}>{Translation[lang].gde_visib}</Text>
                     </View>
                     </TouchableHighlight>
                   </View>
                   <TouchableHighlight onPress={this._onPressEnvironnement}>
                   <View style = {styles.decoupage}>
                     <Image source={require('../img/environnement.png')} style = {styles.logoBloque} />
                     <Text style = {styles.texteCarreau}>{Translation[lang].prot_env}</Text>
                   </View>
                   </TouchableHighlight>
                 </View>
               );
             }
       })()}
       </View>
     );
 }


_onPressValoriserStocks = () =>{
 if(valorize === 'true'){
   valorize = 'false';
 }else{
   valorize ='true';
   price = 'false';
   time ='false';
   visibility ='false';
   environment = 'false';
 }
 this.forceUpdate();
};
_onPressPrix = () =>{
 if(price === 'true'){
   price = 'false';
 }else{
   valorize ='false';
   price = 'true';
   time ='false';
   visibility ='false';
   environment = 'false';
 }
 this.forceUpdate();
};
_onPressTemps = () =>{
 if(time === 'true'){
   time = 'false';
 }else{
   valorize ='false';
   price = 'false';
   time ='true';
   visibility ='false';
   environment = 'false';
 }
 this.forceUpdate();
};
_onPressVisibilite = () =>{
 if(visibility === 'true'){
   visibility = 'false';
 }else{
   valorize ='false';
   price = 'false';
   time ='false';
   visibility ='true';
   environment = 'false';
 }
 this.forceUpdate();
};
_onPressEnvironnement = () =>{
 if(environment === 'true'){
   environment = 'false';
 }else{
   valorize ='false';
   price = 'false';
   time ='false';
   visibility ='false';
   environment = 'true';
 }
 this.forceUpdate();
};
}


export default class Benefits extends Component {
  render() {
    BackAndroid.addEventListener('Back',this.onBackAndroid);
    return (
      <ScrollView>
       <View style={{height:height*0.08}}/>
       <Text style={styles.intro}>{Translation[lang].benef}</Text>
       <BloqueFonction/>
     </ScrollView>
    );
  }
  
  onBackAndroid = () => {
		const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}
const styles = StyleSheet.create({

  intro:{
    textAlign:'center',
    fontSize:20,
    marginTop:height*0.07,
  },
  containerV: {
    alignItems:'center',
    flexDirection: 'column',
    justifyContent: 'center',
    marginTop:width*0.06,
  },
  containerH: {
    flexDirection : 'row',
  },
  decoupage: {
    width: height*0.2,
    height:height*0.2,
    alignItems:'center',
    backgroundColor:'white',
    margin: width*0.05,
  },
  texteCarreau:{
    textAlign : 'center',
    fontWeight: 'bold',
    fontSize: 15,
    marginTop:10,

  },
  logoBloque:{
    width: 40,
    height: 40,
  },

  cliqueDecoupage: {
    width: height*0.38,
    height:height*0.27,
    borderWidth:1,
    borderColor: '#A4D04A',
    backgroundColor:'white',
    alignItems:'center',
  },

  cliqueDecoupageStatique: {
    width: height*0.2,
    height:height*0.2,
    backgroundColor:'white',
    alignItems:'center',
    marginTop:30,
  },

  cliqueTitre: {
    flexDirection : 'row',
    marginTop: 20,

  },
  cliqueTexte: {
    textAlign:'center',
    marginTop: 10,
    fontSize: 15,
  },

});

module.exports = BeneficesEntreprise;
