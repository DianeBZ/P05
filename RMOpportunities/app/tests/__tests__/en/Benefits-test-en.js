// __tests__/Benefits-test-en.js
import 'react-native';
import React from 'react';

global.lang = 'en';
import Benefits from '../../../main/Benefits';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Benefits />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
