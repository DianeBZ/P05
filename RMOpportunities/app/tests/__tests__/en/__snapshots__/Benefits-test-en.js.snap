// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "height": 106.72,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "marginTop": 93.38000000000001,
          "textAlign": "center",
        }
      }
    >
      Benefits for your company
    </Text>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginTop": 45,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "height": 266.8,
                  "margin": 37.5,
                  "width": 266.8,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
              >
                Value your unused stocks
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "height": 266.8,
                  "margin": 37.5,
                  "width": 266.8,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
              >
                Buy at the best rates
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "height": 266.8,
                  "margin": 37.5,
                  "width": 266.8,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
              >
                Save time
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "height": 266.8,
                  "margin": 37.5,
                  "width": 266.8,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
              >
                Benefit from high visibility
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "height": 266.8,
                "margin": 37.5,
                "width": 266.8,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
            >
              Protect the environment
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
